### PR TITLE
Add post conditions to transaction builder helpers

### LINKED ISSUE #17: Add post conditions to transaction builders

### PR DESCRIPTION
**Pull request by yknl**
## Description
This adds the ability to specify post conditions when building STX token transfer, contract deploy and contract function call transactions. The transaction builder API has also been refactored to use an `options` object for some parameters to make it easier to use. Tests related to transaction builder helpers have been moved to a separate file.

resolves #17 

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
Yes, all transaction builder function signatures have been changed.

## Are documentation updates required?
Yes, the readme has been updated.

## Testing information
Generated transactions with post conditions should be tested. 

## Checklist
- [x] Code is commented where needed
- [x] Unit test coverage for new or modified code paths
- [x] `npm run test` passes
- [ ] Changelog is updated
- [x] Tag 1 of @yknl, @zone117x, @reedrosenbluth for review

### CODE REVIEW

**Comment by agraebe** [comment]
I understand this is in the draft - just wanted to confirm: For PRs like this, will we update the documentation?

Would it also make sense to lead with a discussion about the "ideal" method/API/interface for best developer experience and work backwards how it can be implemented? Stuff like this can have a significant impact if I understand correctly :)

**Comment by yknl** [comment]
Yes, PRs like this will include documentation updates. It is one of the checklist items in the pull request template. 

It does make sense to have a discussion about the optimal API. It should've been done at the design stage though. Happy to take feedback into consideration for further improvements.
Part of this refactor was driven by recent feedback from other team members.

File: README.md [hunk 1]
```diff
@@ -172,3 +199,109 @@ If you develop in Typescript, the type checker will help prevent you from creati
 ```typescript
 const listCV = new ListCV([new TrueCV, new IntCV(1)]);
 ```
+
+## Post Conditions
+Three types of post conditions can be added to transactions:
```
Comment (kyranjamie) [inline README.md]: Would be helpful to link to https://github.com/blockstack/stacks-blockchain/blob/master/sip/sip-005-blocks-and-transactions.md#transaction-post-conditions

File: src/builders.ts [hunk 2]
```diff
@@ -8,12 +8,45 @@ import { StacksPrivateKey } from './keys';
...
+ * @param  {Array<PostCondition>} postConditions - an array of post conditions to add to the
+ *                                                  transaction
+ *
+ * @return {StacksTransaction}
+ */
+export interface TokenTransferOptions {
+  nonce?: BigNum;
+  version?: TransactionVersion;
+  memo?: string;
+  postConditions?: Array<PostCondition>;
```
Comment (kyranjamie) [inline src/builders.ts]: Generally I've seen the shorthand array type notation used more often, would be great to stick to one.

Comment (yknl) [reply src/builders.ts]: Is there a way to specify this in the prettier config?

Comment (kyranjamie) [reply src/builders.ts]: No, but we could add a lint rule to the config hank made

**Pull request merged**